### PR TITLE
Det 143 py test to check python files

### PR DESCRIPTION
**Pull request by Kwill3**

### CODE REVIEW

File: src/version_overlap_check.py [hunk 1]
```diff
@@ -94,47 +106,42 @@ def scan_file(filename:str, overlap_list)->int:
...
     # Check for overlaps in this file
     if overlap_count > 0:
         print('List of overlapping versions for keys and their index:', overlap_list)
-        check_failed_flag = True
-        failed_files.append(filename)
+    #     check_failed_flag = True
+    #     failed_files.append(filename)
 
-    return overlap_count
+    return overlap_list
```
Comment (GarethIsaac) [inline src/version_overlap_check.py:140]: You are returning a list, but in the header it says you are returning a int.

**Review by GarethIsaac — CHANGES_REQUESTED** [review]
Please correct the name of the test scenarios and update the information

File: src/version_overlap_check.py (hunk 1 above)
Comment (Kwill3) [reply src/version_overlap_check.py]: Resolved, now returning a tuple of count and list.

File: src/test_version_overlap_check.py [hunk 2]
```diff
@@ -11,7 +11,179 @@
...
     print(test1)
-    assert count == 6
+    assert count == {}
+
+def test_scenario2():
+    """Passing test for first instance of current version
+    """
+    test2 = 0
+    count = scan_file(BASE_PATH + '/tests/test_scenario2.csv', test2)
+    assert count == {}
```
[the remark this answers is not in the document]
Comment (Kwill3) [reply src/test_version_overlap_check.py]: Resolved, now named as output and checking for a tuple [int, dict]

**Comment by Kwill3** [comment]
Descriptions and comments now updated. Return types and tests are also corrected.

**Review by GarethIsaac — APPROVED** [review]

**Pull request merged**